independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/ServerSerialisers.java

```diff
@@ -502,4 +523,15 @@ public static void encodeResponseHeaders(ResteasyReactiveRequestContext requestC
         }
     }
 
+    private static boolean requireSingleHeader(String header) {
+        if (!(header.startsWith(CONTENT) || header.startsWith(CONTENT_LOWER))) {
+            return false;
+        }
+        if (header.length() < CONTENT.length() + 2) {
+            return false;
+        }
+        String substring = header.substring(CONTENT.length() + 1).toLowerCase(Locale.ROOT);
+        return substring.equals(TYPE_LOWER) || substring.equals(LENGTH_LOWER);
+    }
```

Comment:
Not required to merge this PR, but this creates less Strings and performs less comparisons: 

```suggestion
    private static boolean requireSingleHeader(String header) {
        int contentLength = CONTENT.length();
        // Check if starts with "Content"
        if (!header.regionMatches(true, 0, CONTENT, 0, contentLength)) {
            return false;            
        }
        // Check if it ends with "Type" or "Length"
        return header.regionMatches(true, contentLength + 2, TYPE, 0, TYPE.length())
                || header.regionMatches(true, contentLength + 2, LENGTH, 0, LENGTH.length());
    }
```

Reply:
And yet it can be slower when measured, nvm